### PR TITLE
Optional promote and switch to wayland

### PR DESCRIPTION
**Pull request by ccadruvi**
Make promote step optional
Switch to Wayland session when logging in with SDDM

### CODE REVIEW

**Review by qdrop17 — APPROVED** [review]
LGTM

File: build/build.sh [hunk 1]
```diff
@@ -10,24 +10,49 @@ function removeFileIfExists {
...
+    export "$KEY"="$VALUE"
+done
 
-if [[ $# -lt 1 ]]; then
-    echo "Error: No arguments passed. Make sure to pass at least the Netboot IP Address"
+# This argument is required to read the latest kernel version from the netboot server
+if [[ "$netbootIP" == "" ]]; then
+    echo "Error: No arguments passed. Make sure to pass at least the Netboot IP Address, e.g. netbootIP=10.1.30.4"
+    exit 1
 else
```
Comment (SwissGipfel) [inline build/build.sh:25]: This else is not really necessary.
With this else removed, the following if statements wouldn't be nested.

File: build/build.sh [hunk 2]
```diff
@@ -71,5 +102,9 @@ docker rm -f "$squashfsContainerID"
 rm -f "$(pwd)/$tarFileName"
 
 squashfsAbsolutePath="$(pwd)/$squashfsFilename"
-# AzureDevOps specific way of passing an output variable to subsequent steps in the pipeline
-echo "##vso[task.setvariable variable=squashfsAbsolutePath;isOutput=true]$squashfsAbsolutePath"
+
+# If you want to promote the image directly to the caching server, run ./build.sh buildSquashfsAndPromote="true"
```
Comment (SwissGipfel) [inline build/build.sh:106]: Wouldn't 'promoteSquasfs' be clearer? This toggle only controls the promote part. The Squasfs will always be built.

Also, I'd extend this comment a bit like:

> this base squashfs will always be pushed into /dev/

**Review by SwissGipfel — APPROVED** [review]

File: build/build.sh (hunk 2 above)
Comment (ccadruvi) [reply build/build.sh:106]: No, this actually skips the squashfs build, as we don't need it in most cases. This is what saves us time (and the reason to implement this story).

**Pull request merged**